others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,362 @@
+ * @param root head/root node of a tree
+ * @return result that is containing the postorder traversal of a tree
+ */
+std::vector<int64_t> BinaryTree::postOrderIterative(Node *root) {   
+    std::stack<Node *> stack; ///< is used to find and traverse the child nodes.
+    std::vector<int64_t> result;  ///< List of values, sorted in post-order.
+    
+    stack.push(root);
+    
+    while(!stack.empty()) {
+        result.push_back(stack.top()->data);
+        Node *current = stack.top();
+        stack.pop();
+        
+        if(current->left) {
+            stack.push(current->left);
+        }
+        if(current->right) {
+            stack.push(current->right);
+        }
+    }
+
+    reverse(result.begin(), result.end());
+    
+    return result;
+}
+
+/**
+ * @brief inOrderIterative() function that will perform the inorder traversal iteratively,
+ * and return the result array that contain the inorder traversal of a tree.
+ * @param root head/root node of a tree
+ * @return result that is containing the inorder traversal of a tree
+ */
+std::vector<int64_t> BinaryTree::inOrderIterative(Node *root) {
+    std::stack<Node *> stack; ///< is used to find and traverse the child nodes.
+    std::vector<int64_t> result;  ///< List of values, sorted in in-order.
+
+    Node *current = root;
+    
+    while(!stack.empty() || current) {
+        while(current) {
+            stack.push(current);
+            current = current->left;
+        }
+        current = stack.top();
+        stack.pop();    
+        result.push_back(current->data);   
+        current = current->right;
+    }   
+    return result;
+}
+}  //  namespace iterative_tree_traversals
+}  //  namespace others
+
+/**
+ * @brief Test the computed preorder with the actual preorder.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test1(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{1, 2, 4, 5, 3};
+    std::vector<int64_t> result; ///< result stores the preorder traversal of the binary tree
+
+    // Calling preOrderIterative() function by passing a root node,
+    // and storing the preorder traversal in result.
+    result = binaryTree.preOrderIterative(root); 
+
+    // Self-testing the result using `assert`
+    for(int i = 0; i < result.size(); i++)
+        assert(actual_result[i] == result[i]);
+
+    // Printing the result storing preorder.
+    std::cout<< "\nPreOrder Traversal Is : "<< std::endl;
+    for(auto i: result) {
+        std::cout<< i<< "  ";
+    }
+}
+
+/**
+ * @brief Test the computed postorder with the actual postorder.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test2(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{4, 5, 2, 3, 1};

```

Comment:
Please add a one-line description of what the function is for (like the line below).